src/Servers.jl

```diff
@@ -410,108 +118,110 @@ function handle_connection(f::Function, c::Connection;
 
-
-"""
-Create a `HTTP.Stream` and parse the Request headers from a `HTTP.Transaction`.
-If there is a parse error, send an error Response.
-Otherwise, execute stream processing function `f`.
-"""
-function handle_transaction(f::Function, t::Transaction;
-                            final_transaction::Bool=false,
-                            verbose::Bool=false, kw...)
-
-    request = HTTP.Request()
-    http = Streams.Stream(request, t)
-
-    try
-        startread(http)
-    catch e
-        # @show typeof(e)
-        # @show fieldnames(e)
-        if e isa EOFError && isempty(request.method)
-            return
-# FIXME https://github.com/JuliaWeb/HTTP.jl/pull/178#pullrequestreview-92547066
-#        elseif !isopen(http)
-#            @warn "Connection closed"
-#            return
-        elseif e isa HTTP.ParseError
-            @error e
-            status = e.code == :HEADER_SIZE_EXCEEDS_LIMIT  ? 413 : 400
-            write(t, Response(status, body = string(e.code)))
-            close(t)
-            return
-        else
-            rethrow(e)
-        end
-    end
-
-    if verbose
-        @info http.message
-    end
-
-    response = request.response
-    response.status = 200
-    if final_transaction || hasheader(request, "Connection", "close")
-        setheader(response, "Connection" => "close")
-    end
-
-    @async try
-        handle_stream(f, http)
-    catch e
-        if isioerror(e)
-            @warn e
-        else
+struct Server2{S, I}
+    ssl::S
+    server::I
+    hostname::String
+    hostport::String
+end
+Base.isopen(s::Server2) = isopen(s.server)
+Sockets.accept(s::Server2{Nothing, S}) where {S} = Sockets.accept(s.server)
+function getsslcontext(tcp, sslconfig)
+    ssl = SSLContext()
+    setup!(ssl, sslconfig)
+    associate!(ssl, tcp)
+    handshake!(ssl)
+    return ssl
+end
+Sockets.accept(s::Server2) = getsslcontext(accept(s.server), s.ssl)
+Base.close(s::Server2) = close(s.server)
+
+function listenloop(h::Handler, server,
+    tcpisvalid=x->true, connectioncounter=Ref(0),
+    reuse_limit::Int=1, readtimeout::Int=0, verbose::Bool=false)
+    count = 1
+    while isopen(server)
+        try
+            io = accept(server)
+            io === nothing && continue
+            if !tcpisvalid(io)
+                verbose && @info "Accept-Reject:  $io"
+                close(io)
+                continue
+            end
+            connectioncounter[] += 1
+            conn = Connection(server.hostname, server.hostport, 0, 0, true, io)
+            @async begin
+                try
+                    verbose && @info "Accept ($count):  $conn"
+                    handle(h, conn)
+                    verbose && @info "Closed ($count):  $conn"
+                catch e
+                    @error exception=(e, stacktrace(catch_backtrace()))
+                finally
+                    connectioncounter[] -= 1
+                    close(io)
+                    verbose && @info "Closed ($count):  $conn"
```

Comment:
At somepoint we might want all verbosity to be  handled purely by configuring the logger.
But not today